### PR TITLE
Add markerclustererplus w/ git auto-update

### PR DESCRIPTION
**Pull request by hare1039**
Pull request for issue: #9802
Related issue(s): # #

@pvnr0082t Please help me review this PR. Thank you.
Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/mahnunchik/markerclustererplus
 * Official website (optional, not the repository):
 * NPM package url (optional): https://www.npmjs.com/package/markerclustererplus
 * License and its reference: Apache-2.0, ref: https://github.com/mahnunchik/markerclustererplus/blob/master/package.json#L20
 * GitHub / Bitbucket popularity (required):
   - Count of stars: 110
   - Count of watchers: 10
   - Count of forks: 79
 * NPM download stats (optional):
   - Downloads in the last day: 357
   - Downloads in the last week: 2268
   - Downloads in the last month: 7795

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 500 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [ ] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

File: ajax/libs/markerclustererplus/package.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  "keywords": [
+    "google",
+    "maps",
+    "markerclustererplus"
+  ],
+  "repository": {
+    "type": "git",
+    "url": "git://github.com/mahnunchik/markerclustererplus.git"
+  },
+  "homepage": "https://htmlpreview.github.io/?https://github.com/googlemaps/v3-utility-library/blob/master/markerclustererplus/docs/reference.html",
```
Comment (pvnr0082t) [inline ajax/libs/markerclustererplus/package.json]: @hare1039 Where did you get the homepage url? Is it officail homepage?

Comment (hare1039) [inline ajax/libs/markerclustererplus/package.json]: I forgot where I get that url. Sorry about that.
I'll fix my it ASAP.

**Comment by hare1039** [comment]
@pvnr0082t Should `src` folder add into git auto-update? The author wrote the [warning](https://github.com/mahnunchik/markerclustererplus#warning) that there would be no more `src/markerclusterer.min.js` and `src/markerclusterer_packed.js` in the future. What should I do?

**Comment by pvnr0082t** [comment]
@hare1039 Nope. You can keep your auto-update config. Thanks.

**Comment by hare1039** [comment]
@pvnr0082t I've amended the PR. Please review this PR again. Thank you.

**Comment by hare1039** [comment]
@pvnr0082t Do you meen to change the auto-update config to `dist/**/*` and `src/markerclusterer.js` ?

**Comment by pvnr0082t** [comment]
@hare1039 `dist/**/*` and `src/markerclusterer*`

**Comment by hare1039** [comment]
@pvnr0082t I updated the `package.json`. Please review this PR again. Thank you.

**Comment by hare1039** [comment]
@pvnr0082t I removed the map files. Please review this PR again.

File: ajax/libs/markerclustererplus/2.0.15/markerclusterer.min.js [hunk 2]
```diff
@@ -0,0 +1,2 @@
...
+//# sourceMappingURL=markerclusterer.min.js.map
```
Comment (pvnr0082t) [inline ajax/libs/markerclustererplus/2.0.15/markerclusterer.min.js]: @hare1039 It seems like you just remove map files. You should re-minify it.

Comment (hare1039) [inline ajax/libs/markerclustererplus/2.0.15/markerclusterer.min.js]: @pvnr0082t How to minify files using the `web-minify-helper/minify.sh` without creating the `.map` file?

Comment (pvnr0082t) [inline ajax/libs/markerclustererplus/2.0.15/markerclusterer.min.js]: @hare1039 Run `minify.sh` with augment `--no-map`. You can try to read the code in `minify.sh`.

Comment (hare1039) [inline ajax/libs/markerclustererplus/2.0.15/markerclusterer.min.js]: @pvnr0082t OK, I re-minify the old files. Please review this PR again.

**Comment by hare1039** [comment]
@pvnr0082t What can I do for a PR if no other member reviews?

**Comment by pvnr0082t** [comment]
@hare1039 You can try to ping them. ping @cdnjs/intern2

**Comment by maruilian11** [comment]
@PeterDaveHello I think this PR is fine.

**Comment by hare1039** [comment]
@PeterDaveHello I amended the commit message. Please help me reviewing the PR again. Thank you.

**Comment by PeterDaveHello** [comment]
Commit message issues ...
1. The commit message format in the second commit is totally wrong ...
2. `Add`/`add` in the same PR, a little bit old ...
3. **G**it**H**ub not **g**it**h**ub
4. You only add partially old versions ... please clarify that.

**Comment by hare1039** [comment]
@PeterDaveHello I amend the commit massage to
```
Add old versions of markerclustererplus, cc #9802
    Manually add old partial versions due to git auto-update can't get v2.0.15/2.0.16 from npm and different file structure of v2.1.1/2.1.2 from GitHub
```
Is this acceptable?

**Comment by PeterDaveHello** [comment]
@hare1039 I prefer you to discuss with you mentor first.

**Comment by PeterDaveHello** [comment]
The second line of the commit message should always be empty ...

**Comment by hare1039** [comment]
Okey I changed it. Anything missed?

**Comment by hare1039** [comment]
@pvnr0082t I amended it. Please review the commit massage again.

**Comment by PeterDaveHello** [comment]
You should wrap the commit message.

Ref:
https://git-scm.com/book/ch5-2.html
```
Short (50 chars or less) summary of changes

More detailed explanatory text, if necessary.  Wrap it to
about 72 characters or so.  In some contexts, the first
line is treated as the subject of an email and the rest of
the text as the body.  The blank line separating the
summary from the body is critical (unless you omit the body
entirely); tools like rebase can get confused if you run
the two together.

Further paragraphs come after blank lines.

  - Bullet points are okay, too

  - Typically a hyphen or asterisk is used for the bullet,
    preceded by a single space, with blank lines in
    between, but conventions vary here
```

**Comment by hare1039** [comment]
Got it. I fix the commit message. Please review this PR again.

File: ajax/libs/markerclustererplus/package.json [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      {
+        "basePath": "dist",
+        "files": [
+          "**/*"
+        ]
+      },
+      {
+        "basePath": "src",
+        "files": [
+          "markerclusterer*"
```
Comment (PeterDaveHello) [inline ajax/libs/markerclustererplus/package.json]: So just specifically set the rule to src file here.

Comment (hare1039) [inline ajax/libs/markerclustererplus/package.json]: Sorry for the careless mistake I made. I update the rule. Please help me review this PR again. Thank you.

File: ajax/libs/markerclustererplus/package.json [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+      {
+        "basePath": "dist",
+        "files": [
+          "**/*"
+        ]
+      },
+      {
+        "basePath": "src",
+        "files": [
+          "markerclusterer!(*.min.*)"
```
Comment (PeterDaveHello) [inline ajax/libs/markerclustererplus/package.json]: Just write a static rule here to grab the src file, not by excluding `*.min.*`

**Comment by hare1039** [comment]
@pvnr0082t I changed the auto update rule of "src" to static ones. Please review this PR again. Thank you.

File: ajax/libs/markerclustererplus/package.json [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        "basePath": "dist",
+        "files": [
+          "**/*"
+        ]
+      },
+      {
+        "basePath": "src",
+        "files": [
+          "markerclusterer.js",
+          "markerclusterer_packaged.js"
```
Comment (pvnr0082t) [inline ajax/libs/markerclustererplus/package.json]: @hare1039 `"markerclusterer_packaged.js"` -> `"markerclusterer_packed.js "`

**Comment by maruilian11** [comment]
Since `markerclusterer.min.js` and `markerclusterer_packaged.js` have been deprecated and will be removed in the next release.
Maybe we should only fetch `src/markerclusterer.js` in auto-update config.

File: ajax/libs/markerclustererplus/package.json (hunk 5 above)
Comment (hare1039) [inline ajax/libs/markerclustererplus/package.json]: @pvnr0082t Should we only fetch `src/markerclusterer.js` in auto-update config, but manually add the `markerclusterer_packed.js` for the old version as maruilian11 suggest?

**Comment by PeterDaveHello** [comment]
Let's just drop `markerclusterer_packed.js` and `markerclusterer.min.js` from `src`

**Comment by hare1039** [comment]
@pvnr0082t  Shall the old versions' `markerclusterer_packed.js` also be excluded?

**Comment by PeterDaveHello** [comment]
@hare1039 Exclude all.

**Comment by hare1039** [comment]
@pvnr0082t I excluded the `markerclusterer_packed.js`. Please help me review this PR again. Thank you.

**Pull request merged**